src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -40,21 +41,30 @@ def _get_current_date_with_format() -> str:
     return datetime.strftime(datetime.now(), "%Y-%m-%d %H:%M:%S")
 
 
+def _map_result_to_dicts_arr(result: Union[list[ResultProxy], None]):
+    mapped_arr: list[dict] = list[dict]()
+    if result is None:
+        return list()
```

Comment:
Both uses of this function are followed by

```
db_item = dict() if len(mapped_result) == 0 else mapped_result[0]
```

Why not handle that directly in the function?

Reply:
Oh, I see the point now - this function is returning all rows, but in the code you know that you only care about the first result and thus extract it with `[0]`. Makes sense now.

Might be useful to write one line method to encapsulate this, e.g. `_get_first(result_list)` or something like that.

Reply:
✔️